starter_kits/Kotlin/hlt/Entity.kt

```diff
@@ -1,3 +1,22 @@
 package hlt
 
-open class Entity(val owner: PlayerId, val id: EntityId, val position: Position)
\ No newline at end of file
+open class Entity(val owner: PlayerId, val id: EntityId, val position: Position) {
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as Entity
+
+        if (owner != other.owner) return false
+        if (id != other.id) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        var result = owner.hashCode()
+        result = 31 * result + id.hashCode()
```

Comment:
Actually I think you can juse use the integer value of the "id" directly. They are guaranteed to be unique across all the owners and so on. The only "difficult" one is the shipyard - that always has id of -1. So in that case you can return owner*(-1) for example.

Reply:
(But obviously what you have works too and I'm ok with it as-is.)

Reply:
ok, thanks.